tests/run/binop_reverse_methods_GH2056.pyx

```diff
@@ -4,29 +4,55 @@ cimport cython
 @cython.cclass
 class Base(object):
     """
-    >>> Base() + 2
+    >>> Base(1) + 2
     'Base.__add__(Base(), 2)'
-    >>> 2 + Base()
+    >>> 2 + Base(1)
     'Base.__radd__(Base(), 2)'
 
-    >>> Base() ** 2
+    >>> Base(0) + 2  #doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    ...
+    TypeError: unsupported operand type...
+    >>> 2 + Base(0)  #doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    ...
+    TypeError: unsupported operand type...
+
+    >>> Base(1) ** 2
     'Base.__pow__(Base(), 2, None)'
-    >>> 2 ** Base()
+    >>> 2 ** Base(1)
     'Base.__rpow__(Base(), 2, None)'
-    >>> pow(Base(), 2, 100)
+    >>> pow(Base(1), 2, 100)
     'Base.__pow__(Base(), 2, 100)'
     """
+    implemented: cython.int
+
+    def __init__(self, implemented):
+        self.implemented = implemented
```

Comment:
I wonder how to make the usage readable in the tests. What about his?
```suggestion
    def __init__(self, implemented="implemented"):
        assert implemented in ("implemented", "not implemented")
        self.implemented = implemented == "implemented"
```

Reply:
I've changed it to be a keyword-only argument, which I think makes it more readable.